plugin/gc-app/src/main/java/io/deephaven/app/GcNotificationConsumer.java

```diff
@@ -0,0 +1,114 @@
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryUsage;
+import java.util.Arrays;
+import java.util.Objects;
+
+final class GcNotificationConsumer {
+
+    private static final TableDefinition DEFINITION = TableDefinition.of(
+            ColumnDefinition.ofLong("Id"),
+            ColumnDefinition.ofTime("Start"),
+            ColumnDefinition.ofTime("End"),
+            ColumnDefinition.ofString("GcName"),
+            ColumnDefinition.ofString("GcAction"),
+            ColumnDefinition.ofString("GcCause"),
+            ColumnDefinition.ofLong("Reclaimed"));
+    private static final int CHUNK_SIZE = ArrayBackedColumnSource.BLOCK_SIZE;
+
+    public static TableDefinition definition() {
+        return DEFINITION;
+    }
+
+    public static Table stats(Table notificationInfo) {
+        // Would be great to have a aggBy Exponential Decayed Sum so we could have "GC rate over the last 1, 5, 15 min"
+        // Could further extend with updateBy Exponential Decayed Sum and graphs of rates.
+        return notificationInfo
+                .updateView("Duration=(End-Start)/1000000000")
+                .aggBy(Arrays.asList(
+                        Aggregation.AggCount("Count"),
+                        Aggregation.AggSum("DurationTotal=Duration", "ReclaimedTotal=Reclaimed"),
+                        Aggregation.AggMax("DurationMax=Duration"),
+                        Aggregation.AggAvg("DurationAvg=Duration", "ReclaimedAvg=Reclaimed"),
+                        Aggregation.AggApproxPct("Reclaimed",
+                                PercentileOutput.of(0.5, "ReclaimedP_50")),
+                        Aggregation.AggApproxPct("Duration",
+                                PercentileOutput.of(0.5, "DurationP_50"),
+                                PercentileOutput.of(0.9, "DurationP_90"),
+                                PercentileOutput.of(0.95, "DurationP_95"),
+                                PercentileOutput.of(0.99, "DurationP_99")),
+                        Aggregation.AggLast(
+                                "LastId=Id",
+                                "LastStart=Start",
+                                "LastEnd=End",
+                                "LastReclaimed=Reclaimed")),
+                        "GcName", "GcAction", "GcCause");
+    }
+
+    private final StreamConsumer consumer;
+    private final long vmStartMillis;
+    private WritableChunk<Values>[] chunks;
+
+    GcNotificationConsumer(StreamConsumer consumer) {
+        this.consumer = Objects.requireNonNull(consumer);
+        this.vmStartMillis = ManagementFactory.getRuntimeMXBean().getStartTime();
+        // noinspection unchecked
+        chunks = StreamToTableAdapter.makeChunksForDefinition(DEFINITION, CHUNK_SIZE);
+    }
+
+    public synchronized void add(GarbageCollectionNotificationInfo gcNotification) {
+        final GcInfo gcInfo = gcNotification.getGcInfo();
+        chunks[0].asWritableLongChunk().add(gcInfo.getId());
```

Comment:
Could be tighter if you used `set` instead of `add`, keeping a single counter instead of one per column. We could also dump the casts if we really wanted.